src/services/expenses.service.js

```diff
@@ -0,0 +1,76 @@
+let expenses = [];
+let nextId = 1;
+
+const getAllExpenses = () => {
+  return expenses;
+};
+
+const getById = (id) => {
+  return expenses.find((exp) => exp.id === parseInt(id));
+};
+
+const normalizeCategory = (categories) => {
+  return Array.isArray(categories) || !categories ? categories : [categories];
+};
+
+const filterExpensesById = (id) => {
+  expenses = expenses.filter((exp) => exp.userId === parseInt(id));
```

Comment:
You are reassigning the `expenses` array with a filtered version of itself. This is a mutation that can lead to unexpected behavior in your code. It would be safer to return a new array with the filtered expenses instead of modifying the original array.